proposals/0026-hlsl-long-vector-type.md

```diff
@@ -129,61 +154,60 @@ Refer to the HLSL spec for an exhaustive list of [Operators](https://learn.micro
+Representation of native vectors in DXIL depends on [dxil vectors](NNNN-dxil-vectors.md).
+
 ### Debug Support
 
-First class debug support for HLSL vectors. Emit `llvm.dbg.declare` and `llvm.dbg.value` intrinsics that can be used by tools for better debugging experience. Open Issue: Handle DXIL scalarized and vector paths.
+First class debug support for HLSL vectors. Emit `llvm.dbg.declare` and `llvm.dbg.value` intrinsics that can be used by tools for better debugging experience.
+These should enable tracking vectors through their scalarized and native vector usages.
 
 ### Diagnostic Changes
 
 Error messages should be produced for use of long vectors in unsupported interfaces.
 
-* The shader signature.
-* A cbuffer/tbuffer.
-* A work graph record.
-* A mesh or ray tracing payload.
+* Typed buffer element types.
+* Parameters to the entry function.
+* Return types from the entry function.
+* Cbuffers blocks.
+* Cbuffers global variables.
+* Tbuffers.
+* Work graph records.
+* Mesh/amplification payload entry parameter structures.
+* Ray tracing `Payload` parameter structures used in `TraceRay` and `anyhit`/`closesthit`/`miss` entry functions.
+* Ray tracing `Parameter` parameter structures used in `CallShader` and `callable` entry functions.
+* Ray tracing `Attributes` parameter structures used in `ReportHit` and `closesthit` entry functions.
 
 Errors should also be produced when long vectors are used as parameters to intrinsics
  with vector parameters of variable length, but aren't permitted as listed in [Disallowed vector intrinsics](#disallowed-vector-intrinsics)
 Attempting to use any swizzle member-style accessors on long vectors should produce an error.
-Declaring vectors of length longer than 128 should produce an error.
+Declaring vectors of length longer than 1024 should produce an error.
 
 ### Validation Changes
 
 Validation should produce errors when a long vector is found in:
 
 * The shader signature.
 * A cbuffer/tbuffer.
-* A work graph record.
-* A mesh or ray tracing payload.
+* Work graph records.
+* Mesh/amplification payload entry parameter structures.
+* Ray tracing `Payload` parameter structures used in `TraceRay` and `anyhit`/`closesthit`/`miss` entry functions.
+* Ray tracing `Parameter` parameter structures used in `CallShader` and `callable` entry functions.
+* Ray tracing `Attributes` parameter structures used in `ReportHit` and `closesthit` entry functions.
+* Metadata
 
 Use of long vectors in unsupported intrinsics should produce validation errors.
 
-## Runtime Additions
-
-Support for Long vectors requires dxil vector support as defined in [the specification](NNNN-dxil-vectors.md).
+### Device Capability
 
-Use of long vectors in a shader should be indicated in DXIL with the corresponding
- shader model version and shader feature flag.
+Devices that support Shader Model 6.9 will be required to fully support this feature.
```

Comment:
In the interchange format section above we have:

> Long vectors can be represented in DXIL, SPIR-V or other interchange formats as scalarized elements or native vectors.
Representation of native vectors in DXIL depends on [dxil vectors](NNNN-dxil-vectors.md).

This seems to imply that long vectors can be supported in existing shader models.  I think it's only native DXIL vectors feature that actually requires SM 6.9?

Reply:
That wasn't what I intended to imply with that statement, rather that implementations could choose either approach as we intend to do at least temporarily. However, it is true that this is possible. One of the major remaining questions is if we should.

Reply:
Before I resolve this conversation: do we have something written down that is tracking this remaining question?

Reply:
I've created #371